File: c/04-vetores/main080.c
/*Faça um algoritmo que preencha um vetor de 30 posições com números entre 1 e 
15 sorteados pelo computador. Depois disso, peça para o usuário digitar um 
número (chave) e seu programa deve mostrar em que posições essa chave foi 
encontrada. Mostre também quantas vezes a chave foi sorteada.

Data 02/11/2025 -> 40ª Dia
*/

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(){

    int posicoes[30];
    int vetor[30];
    int cont = 0;

    srand(time(NULL));
    for(int i = 0; i < 30; i++){

        vetor[i] = rand() % 15 + 1;
    }
    int chave;
    printf("Digite a chave: ");
    scanf("%d",&chave);

    for(int i = 0; i < 30; i++){

        if(vetor[i] == chave){

            posicoes[cont] = i;
            cont ++;
        }
    }
    if(cont > 0) {
        printf("Chave %d encontrada %d vezes nas posicoes: ", chave, cont);
        for(int i = 0; i < cont; i++) {
            printf("%d ", posicoes[i]);
        }
    } else {
        printf("Chave %d nao encontrada", chave);
    }
    printf("\nVetor sorteado: ");
    for(int i = 0; i < 30; i++){

        printf("%d ",vetor[i]);
    }
    return 0;
}

File: python/04-lista/main080.py
''' Faça um algoritmo que preencha um vetor de 30 posições com números entre 1 e 
15 sorteados pelo computador. Depois disso, peça para o usuário digitar um 
número (chave) e seu programa deve mostrar em que posições essa chave foi 
encontrada. Mostre também quantas vezes a chave foi sorteada.

Data 02/11/2025 -> 40ª Dia
'''

import random

lista = []
posicoes = []
cont = 0


for i in range(30):
    aleatorio = random.randint(1,15)
    lista.append(aleatorio)

chave = int(input("Digite a chave: "))
for i in range(30):
    if lista[i] == chave:
        posicoes.append(i) 
        cont += 1
if cont > 0:
    print(f'A chave {chave} foi encontrada {cont} vezes')
    print(f'Posições: {posicoes}')
else:
    print(f'A chave {chave} não foi encontrada')

print(f'Vetor sorteado: {lista}')